### DISCUSSION TITLE
If I have an index.html, I cannot get 404?

### DISCUSSION THREAD

**Discussion by mdnahas** [pocketbase/pocketbase#3767] [Q&A] [answered]
I ran into a weird behavior.   If I have no files in pb_public and point my browser to http://127.0.0.1:8090/asdfasdf, I get a 404 error.  That I expected.  But if I put a single file, index.html, in pb_public and point my browser to http://127.0.0.1:8090/asdfasdf, I get the served index.html.  The same happens with http://127.0.0.1:8090/asdfasdf/asdf.

So, if index.html is present, it seems any bad URL causes index.html to be served.  It doesn't generate a 404 error.  That seems wrong to me.

I'm running pocketbase version 0.19.3 on Ubuntu 20.04.6 LTS.

**Comment by ganigeorgiev** [comment, marked as answer]
This is intentional to support SPA that uses pretty urls.

You can disable this behavior with the `--indexFallback` flag:

```sh
# for all flags start with --help
./pocketbase serve --indexFallback=0
```

**Reply by mdnahas** [reply]
Thanks.  Weird, but understandable given your target market.

**Reply by ganigeorgiev** [reply]
Note that we currently don't have a default 404 HTML page (_this may change in the future_), and from user perspective it may be better to be redirected to the homepage instead of seeing the 404 JSON response.

**Reply by mdnahas** [reply]
Thanks for the mind-reading --- I was just checking if you had a default 404 page!

**Reply by jockm** [reply]
I would really like to have a hook for 404 handing.  That way I could choose to return a 404 page, or log attempts, do something clever to get them where they want to go, etc
